subprojects/build-cache/src/main/java/org/gradle/caching/local/internal/DefaultBuildCacheTempFileStore.java

```diff
@@ -39,7 +39,7 @@ public void withTempFile(BuildCacheKey key, Action<? super File> action) {
         File tempFile = null;
         try {
             try {
-                tempFile = File.createTempFile(hashCode, PARTIAL_FILE_SUFFIX, dir);
+                tempFile = File.createTempFile(hashCode + "-", PARTIAL_FILE_SUFFIX, dir);
```

Comment:
Why should we add the `-` after the `hashCode`? Wouldn't the file then be called `0001c6a8b349b2cb706c2037d331c52f-.part` instead of `0001c6a8b349b2cb706c2037d331c52f.part`?

Reply:
`createTempFile()` adds a timestamp, so the file name now looks like `0001c6a8b349b2cb706c2037d331c52f-13454389536534.part` instead of `0001c6a8b349b2cb706c2037d331c52f13454389536534.part`.